arcgis-ios-sdk-samples/Maps/Download preplanned map/DownloadPreplannedMapViewController.swift

```diff
@@ -0,0 +1,243 @@
+
+        activityIndicatorView.startAnimating()
+
+        onlineMap.load(completion: { [weak self] (error) in
+            self?.activityIndicatorView.stopAnimating()
+
+            if let error = error {
+                print("Error encountered loading the map : \(error.localizedDescription)")
+                return
+            }
+
+            self?.loadPreplannedMapAreas()
+        })
+        self.onlineMap = onlineMap
+    }
+
+    private func loadPortal() {
+        guard portal.loadStatus != .loaded else { return }
+
+        let portalItem = AGSPortalItem(portal: portal, itemID: Constants.portalItemIdentifier)
+        self.portalItem = portalItem
+
+        activityIndicatorView.startAnimating()
+
+        portal.load { [weak self] (error) in
+            self?.activityIndicatorView.stopAnimating()
+
+            if let error = error {
+                print("Error encountered loading the portal : \(error.localizedDescription)")
+                return
+            }
+
+            self?.loadOnlineMap()
+            self?.retrieveAvailablePreplannedMapAreas()
+        }
+    }
+
+    private func loadPreplannedMapAreas() {
+        remoteAvailablePreplannedMapAreas.forEach { area in
+            activityIndicatorView.startAnimating()
+
+            area.load { [weak self, unowned area] (error) in
+                self?.activityIndicatorView.stopAnimating()
+
+                if let error = error {
+                    print("Error encountered loading the area : \(error.localizedDescription)")
+                    return
+                }
+
+                self?.remoteLoadedPreplannedMapAreas.append(area)
+            }
+        }
+    }
+
+    private func removeDownloadedMapPackages() {
+        mapView.map = onlineMap
+
+        DispatchQueue.global(qos: .userInitiated).async { [weak self] in
+            self?.uniqueLocalMapPackages.forEach { package in
+                let path = package.fileURL.path
+                if FileManager.default.fileExists(atPath: path) {
+                    try? FileManager.default.removeItem(atPath: path)
+                }
+            }
+            self?.uniqueLocalMapPackages.removeAll()
+
+            DispatchQueue.main.async {
+                self?.updateView()
+            }
+        }
+    }
+
+    private func retrieveAvailablePreplannedMapAreas() {
+        guard let portalItem = portalItem else { return }
+
+        activityIndicatorView.startAnimating()
+
+        cancelable?.cancel()
+        offlineTask = AGSOfflineMapTask(portalItem: portalItem)
+
+        cancelable = offlineTask?.getPreplannedMapAreas { [weak self] (preplannedAreas, error) in
+            self?.activityIndicatorView.stopAnimating()
+
+            if let error = error {
+                print("Error encountered loading preplanned map areas : \(error.localizedDescription)")
+                return
+            }
+
+            self?.remoteAvailablePreplannedMapAreas = preplannedAreas ?? []
```

Comment:
The order in which the list is populated with the map areas is random. Is this expected?

Reply:
While the other samples don't enforce a strict order, I agree that we can do better here. I pushed a commit that will alphabetize them by area title : [b610186d](https://github.com/Esri/arcgis-runtime-samples-ios/pull/758/commits/b610186d54979bd084974573696c41e4105359c1).